TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/bot/BotActivity.java

```diff
@@ -201,10 +208,11 @@ public void onServiceDisconnected(ComponentName componentName) {
     };
 
     @Override
-    public void onConnect(String room) {
+    public void onConnect(String room, String serverAddress) {
         debugView.showPermanently(getString(R.string.connected));
         switchableView.setDisplayedChild(1);
 
+        lastServerPreferences.saveLastConnectedServer(serverAddress);
```

Comment:
💯 This should live in the presenter ideally. We should create an interface, because all of the presenters live in core and preferences is an implementation detail based on the environment, in this case, Android.  The presenter can then be responsible for telling the views etc of the initial server address and passing it off to the data layer. Thanks for mentioning this @blundell I didn't pick up on it, been too long away from this.